src/sparseml/pytorch/torchvision/train.py

```diff
@@ -432,21 +434,52 @@ def collate_fn(batch):
         alpha = 1.0 - args.model_ema_decay
         alpha = min(1.0, alpha * adjust)
         model_ema = utils.ExponentialMovingAverage(
-            model_without_ddp, device=device, decay=1.0 - alpha
+            model, device=device, decay=1.0 - alpha
         )
 
-    if args.resume:
-        checkpoint = torch.load(args.resume, map_location="cpu")
-        model_without_ddp.load_state_dict(checkpoint["model"])
-        if not args.test_only:
-            optimizer.load_state_dict(checkpoint["optimizer"])
-            lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
-        args.start_epoch = checkpoint["epoch"] + 1
+    if args.checkpoint_path:
+        checkpoint = _load_checkpoint(args.checkpoint_path)
+
+        # restore state from prior recipe
+        manager = ScheduledModifierManager.from_yaml(args.recipe_path)
+        checkpoint_manager = ScheduledModifierManager.from_yaml(
+            checkpoint["checkpoint_recipe"]
+        )
+        checkpoint_manager.apply_structure(model, epoch=checkpoint["epoch"])
+
+        # load values
+        model.load_state_dict(checkpoint["model"])
+        optimizer.load_state_dict(checkpoint["optimizer"])
+        lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
```

Comment:
any reason these can't be shared between the two cases? Also thoughts on not resuming lr_scheduler state for a fresh run? I think optimizer/ema/scaler are ok

Reply:
Pulled out into separate block